transport/http2_server.go

```diff
@@ -202,7 +206,10 @@ func (t *http2Server) operateHeaders(hDec *hpackDecoder, s *Stream, frame header
 		recv: s.buf,
 	}
 	s.method = hDec.state.method
-
+	if t.tracing {
+		s.tr = trace.New("grpc.Recv."+MethodFamily(s.method), s.method)
```

Comment:
This is now pushing RPC-specific knowledge into the transport package that was otherwise fairly agnostic on the matter.